src/Illuminate/Database/Eloquent/Factories/Factory.php

```diff
@@ -235,11 +235,15 @@ public function createOneQuietly($attributes = [])
     /**
      * Create a collection of models and persist them to the database.
      *
-     * @param  int|iterable<int, array<string, mixed>>  $records
+     * @param  int|null|iterable<int, array<string, mixed>>  $records
      * @return \Illuminate\Database\Eloquent\Collection<int, \Illuminate\Database\Eloquent\Model|TModel>
      */
-    public function createMany(int|iterable $records)
+    public function createMany(int|null|iterable $records = null)
```

Comment:
`null` is a PHP 8.2 feature. We can't add this unless breaking 8.1 support.

Reply:
Is that true? Why is this passing the 8.1 test suite? [Nullable types where added in 7.1](https://www.php.net/manual/en/migration71.new-features.php) and then using `null` with the union syntax [was added in PHP 8.0](https://php.watch/versions/8.0/union-types) with the rest of the union types.

Am I missing something here?

Reply:
@JHWelch that's the `?` sign. The "type" `null` was added in PHP 8.2: https://www.php.net/releases/8.2/en.php#null_false_true_types

Reply:
Looking at the link you sent, this is a `null` _standalone_ types.

You can see the example above with the PHP <8.2 code snippet there is a `null` _union_ type.

Reply:
@driesvints the left here is what I'm doing

![Screenshot_20230815-074230.png](https://github.com/laravel/framework/assets/4480375/50be2d9f-dd70-4ac6-a78e-c42fcf2ff6fe)

Reply:
Hah, that works indeed. Learned something new today 🙂  thanks!

Reply:
@driesvints In 8.2 you can use `null` as a standalone type, in 8.1 you can still use it as a union type like the above function. In the snippet below you would only encounter a fatal error when reaching the Bar function.

```php
<?php

function Foo(int|null $text = null): string|null {
    return 'test';
}

function Bar(null $text = null): null {
    return null;
}
```